Fix lodash.get to exclude undefined if a default value is provided

jacobgardner opened the pull request:
Quick example of the issue: 

```
interface SomeInterface {
    optional?: string;
}

const obj: SomeInterface = getSomeInterface();

const value = _.get(obj, 'optional', 'defaultValue');
// ^ value should be of type string since we provided a default value, but it is `string | undefined`
```

This PR updates `lodash.get` to exclude `undefined` when a default value is provided.

To use `Exclude`, we had to update typescript version to 2.8.  The file with the relevant change is `types/lodash/common/object.d.ts`


- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://lodash.com/docs/4.17.11#get
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

todd commented:
Speaking strictly for the package I contribute to (Sequelize), I'm not sure A) why this change is included in this PR and B) why that change is necessary in the first place. The Sequelize website is a much better resource than the GitHub repo.

jacobgardner commented:
@todd That change was in response to TRAVIS-CI failing with this error. 

```
Error in sequelize
Error: At least one of the project urls listed in the header, ["http://sequelizejs.com"], must match the homepage listed by npm, 'https://github.com/sequelize/sequelize'.
If your d.ts file is not for the npm package with URL https://github.com/sequelize/sequelize,
change the name by adding -browser to the end and change the first line
of the Definitely Typed header to
    // Type definitions for non-npm package sequelize-browser
```

Here's the link to that build failure: https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/493408858?utm_source=github_status&utm_medium=notification

todd commented:
@jacobgardner Thanks for the context. Seems curious to me that that started failing all of the sudden. I reached out on Gitter to make sure this change was intentional.

aj-r commented:
@todd  @jacobgardner Maybe you could put both URLs in the header? That error message suggests that you can put multiple URLs.

Anyways, the main change here is that we're dropping support for TS 2.6 and TS 2.7 - the reason being that `lodash` needs to use a TS 2.8 feature.

jacobgardner commented:
@todd Both URLs are now in the sequelize project and travis seems ok with that. 

@aj-r Thanks for the help. I've updated the wrappers, added tests, and run generate so I think we're good now.

todd commented:
@jacobgardner Looks good from my perspective - I'll let one of the lodash contributors approve this PR since that's the principal package being changed.

aj-r commented on types/lodash/lodash-tests.ts:
```diff
@@ -5142,6 +5142,9 @@ fp.now(); // $ExpectType number
 // _.get
 {
+    const value: string | undefined = anything;
+    const defValue: boolean = anything;
```
nit: I'd prefer `defaultValue` (avoid abbreviations if possible)

jacobgardner commented:
@typicode @niieani - Just pinging to see if either of you have issues with this PR for lowdb or if you know who should sign off

@ssddi456 - Pinging you for yog2-kernel approval.
